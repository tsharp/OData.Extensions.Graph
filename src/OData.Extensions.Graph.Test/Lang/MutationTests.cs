using Microsoft.OData;
using OData.Extensions.Graph.Lang;
using Snapshooter.Xunit;
using Xunit;

namespace OData.Extensions.Graph.Test.Lang
{
    public class MutationTests
    {
        [Fact]
        public static void SimpleUpdate()
        {
            // Arrange
            var translator = new OperationTranslator(DebugBindingResolver.Instance, Common.GetEdmModel());

            // Act
            var filerByUserId = translator.TranslateMutation("/user?$select=Status", "Id");

            // Assert
            filerByUserId.DocumentNode.ToString(true).MatchSnapshot();
        }

        [Fact]
        public static void SimpleUpdateWithFilter ()
        {
            // Arrange
            var translator = new OperationTranslator(DebugBindingResolver.Instance, Common.GetEdmModel());

            // Act & Assert
            Assert.Throws<ODataException>(() => translator.TranslateMutation("/user?$select=Status&$filter=Id eq null", "Id"));
        }
    }
}
