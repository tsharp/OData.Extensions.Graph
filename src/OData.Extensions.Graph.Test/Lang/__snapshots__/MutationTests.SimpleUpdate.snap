mutation {
  user(Id: $Id) {
    Status
  }
}
